tensorboard/webapp/deeplink/hash.ts

```diff
@@ -23,25 +23,28 @@ const TAB = '__tab__';
 
 @Injectable()
 export class HashDeepLinker implements DeepLinkerInterface {
-  private readonly tfStorage: TfStorageElement;
+  tfStorage: DeepLinkerInterface & {
+    migrateLegacyURLScheme: () => void;
+  };
 
   constructor() {
-    this.tfStorage = document.createElement('tf-storage');
-    const tfGlobals = document.createElement('tf-globals');
+    document.createElement('tf-storage');
+    document.createElement('tf-globals');
+    this.tfStorage = (window as any)['tensorboard']['tf_storage'];
```

Comment:
I don't want to require it for this work but, out of curiosity, could this be cleaner with the usage of an externs file that describes the public API?